src/Microsoft.Data/DataFrame.cs

```diff
@@ -34,6 +35,132 @@ public DataFrame(IList<BaseColumn> columns)
+                        dataFrameColumn = new ArrowStringColumn(field.Name, dataBuffer, offsetBuffer, nullBuffer, stringArray.Length);
+                        break;
+                    case ArrowTypeId.UInt8:
+                    case ArrowTypeId.UInt16:
+                    case ArrowTypeId.UInt32:
+                        ReadOnlyMemory<byte> uintValueBuffer = (arrowArray as PrimitiveArray<uint>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> uintNullBitMapBuffer = (arrowArray as PrimitiveArray<uint>).NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<uint>(field.Name, uintValueBuffer, uintNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.UInt64:
+                        ReadOnlyMemory<byte> ulongValueBuffer = (arrowArray as PrimitiveArray<ulong>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> ulongNullBitMapBuffer = (arrowArray as PrimitiveArray<ulong>).NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<ulong>(field.Name, ulongValueBuffer, ulongNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.Binary:
+                    case ArrowTypeId.Date32:
+                    case ArrowTypeId.Date64:
+                    case ArrowTypeId.Dictionary:
+                    case ArrowTypeId.FixedSizedBinary:
+                    case ArrowTypeId.HalfFloat:
+                    case ArrowTypeId.Interval:
+                    case ArrowTypeId.List:
+                    case ArrowTypeId.Map:
+                    case ArrowTypeId.Null:
+                    case ArrowTypeId.Struct:
+                    case ArrowTypeId.Time32:
+                    case ArrowTypeId.Time64:
+                    default:
+                        throw new NotImplementedException(nameof(fieldType.Name));
+                }
+                _table.InsertColumn(ColumnCount, dataFrameColumn);
+                fieldIndex++;
+            }
+        }
+
+        public IEnumerable<RecordBatch> AsArrowRecordBatches()
+        {
+            Apache.Arrow.Schema.Builder schemaBuilder = new Apache.Arrow.Schema.Builder();
+
+            int columnCount = ColumnCount;
+            for (int i = 0; i < columnCount; i++)
+            {
+                BaseColumn column = Column(i);
+                Field field = column.Field;
+                schemaBuilder.Field(field);
+            }
+
+            Schema schema = schemaBuilder.Build();
+            List<Apache.Arrow.Array> arrays = new List<Apache.Arrow.Array>();
+
+            int recordBatchLength = Int32.MaxValue;
+            int numberOfRowsInThisRecordBatch = (int)Math.Min(recordBatchLength, RowCount);
+            long numberOfRowsProcessed = 0;
+
+            while (numberOfRowsProcessed < RowCount || RowCount == 0)
```

Comment:
You may want to add a comment for the `RowCount == 0` clause. I don't think it is obvious why this is here.